server/services/recategorisationFilter.ts

```diff
@@ -0,0 +1,161 @@
+
+export const LOW_RISK_OF_ESCAPE = 'lowRiskOfEscape'
+const LOW_ROSH = 'lowRosh'
+export const NO_CURRENT_TERRORISM_OFFENCES = 'noCurrentTerrorismOffences'
+export const NO_ROTL_RESTRICTIONS_OR_SUSPENSIONS = 'noRotlRestrictionsOrSuspensions'
+export const NOT_MARKED_AS_NOT_FOR_RELEASE = 'notMarkedAsNotForRelease'
+export const STANDARD_OR_ENHANCED_INCENTIVE_LEVEL = 'standardOrEnhancedIncentiveLevel'
+export const TIME_LEFT_TO_SERVE_BETWEEN_12_WEEKS_AND_3_YEARS = 'timeLeftToServeBetween12WeeksAnd3Years'
+export const NO_ADJUDICATIONS_IN_THE_LAST_3_MONTHS = 'noAdjudicationsInTheLastThreeMonths'
+
+interface RecategorisationHomeFilters {
+  suitabilityForOpenConditions: Array<
+    | typeof LOW_RISK_OF_ESCAPE
+    | typeof LOW_ROSH
+    | typeof NO_CURRENT_TERRORISM_OFFENCES
+    | typeof NO_ROTL_RESTRICTIONS_OR_SUSPENSIONS
+    | typeof NOT_MARKED_AS_NOT_FOR_RELEASE
+    | typeof STANDARD_OR_ENHANCED_INCENTIVE_LEVEL
+    | typeof TIME_LEFT_TO_SERVE_BETWEEN_12_WEEKS_AND_3_YEARS
+    | typeof NO_ADJUDICATIONS_IN_THE_LAST_3_MONTHS
+  >
+}
+
+export const recategorisationHomeFilters = {
+  suitabilityForOpenConditions: {
+    [LOW_ROSH]: 'Low RoSH',
+    [LOW_RISK_OF_ESCAPE]: 'Low risk of escape',
+    [NO_CURRENT_TERRORISM_OFFENCES]: 'No current terrorism offences',
+    [NO_ROTL_RESTRICTIONS_OR_SUSPENSIONS]: 'No ROTL restrictions or suspensions',
+    [NO_ADJUDICATIONS_IN_THE_LAST_3_MONTHS]: 'No adjudications in the last 3 months',
+    [NOT_MARKED_AS_NOT_FOR_RELEASE]: "Not marked as 'Not for release'",
+    [STANDARD_OR_ENHANCED_INCENTIVE_LEVEL]: 'Standard or Enhanced incentive level',
+    [TIME_LEFT_TO_SERVE_BETWEEN_12_WEEKS_AND_3_YEARS]: 'Time left to serve is between 12 weeks and 3 years',
+  },
+}
+
+const dateIsNotBetween12WeeksAnd3Years = (dateString: string): boolean =>
+  moment(dateString).isAfter(moment().add(3, 'years')) || moment(dateString).isBefore(moment().add(12, 'weeks'))
+
+const getDateNMonthsAgo = (n: number) => {
+  return moment().subtract(n, 'month').format('YYYY-MM-DD')
+}
+
+const loadAdjudicationsData = async (
+  prisoners,
+  nomisClient,
+  agencyId: string
+): Promise<NomisAdjudicationHearingDto[]> => {
+  const prisonerNumbers = prisoners.map(prisoner => prisoner.offenderNo)
+  const dateThreeMonthsAgo = getDateNMonthsAgo(3)
+  const dateTwoMonthsAgo = getDateNMonthsAgo(2)
+  const dateOneMonthAgo = getDateNMonthsAgo(1)
+  const [adjudicationsThreeMonthsAgo, adjudicationsTwoMonthsAgo, adjudicationsLastMonth] = await Promise.all([
+    nomisClient.getOffenderAdjudications(prisonerNumbers, dateThreeMonthsAgo, dateTwoMonthsAgo, agencyId),
+    nomisClient.getOffenderAdjudications(prisonerNumbers, dateTwoMonthsAgo, dateOneMonthAgo, agencyId),
+    nomisClient.getOffenderAdjudications(prisonerNumbers, dateOneMonthAgo, moment().format('YYYY-MM-DD'), agencyId),
+  ])
+  return [...adjudicationsThreeMonthsAgo, ...adjudicationsTwoMonthsAgo, ...adjudicationsLastMonth]
```

Comment:
i don't follow this - so may very well have missed something

is this possibly simplified with a single call from 

` nomisClient.getOffenderAdjudications(prisonerNumbers, dateThreeMonthsAgo, moment().format('YYYY-MM-DD'), agencyId),`